File: web-sources/s8-web-front/svg/WebSVG_Element.js
import { WebSVG_Canvas } from "svg/WebSVG_Canvas.js";
import { NeObject } from "/s8-io-bohr-neon/NeObject.js";




/**
 * 
 */
export class WebSVG_Element extends NeObject {

    /**
     * @type{WebSVG_Canvas}
     */
    canvas;


    /**
     * @type{boolean}
     */
    isBoundingBoxRelevant;


    /**
     * @type{SVGElement}
     */
    SVG_node;

    
    constructor(){
        super();
    }

    /**
     * 
     * @param {WebSVG_Canvas} canvas 
     */
    link(canvas){
        this.canvas = canvas;
    }



	S8_set_isBoundingBoxRelevant(state) {
		this.isBoundingBoxRelevant = state;
	}
	
	
	/**
	 * 
	 * @param solidity
	 */
	public void setStrokeSolidity(WebSVG_StrokeSolidity solidity) {
		vertex.setUInt8Field("stroke-solidity", solidity.code);
	}
	
	
	/**
	 * 
	 * @param color
	 */
	public void setStrokeColor(WebSVG_StrokeColor color) {
		vertex.setUInt8Field("stroke-color", color.code);
	}
	
	
	/**
	 * Style: 
	 * @param thickness
	 */
	public void setStrokeThickness(WebSVG_StrokeThickness thickness) {
		vertex.setUInt8Field("stroke-thickness", thickness.code);
	}

    




}

File: sources/com/s8/web/front/websvg/WebSVG_Element.java
package com.s8.web.front.websvg;

import com.s8.io.bohr.neon.core.NeBranch;
import com.s8.io.bohr.neon.core.NeObject;

public class WebSVG_Element extends NeObject {

	public WebSVG_Element(NeBranch branch, String typeName) {
		super(branch, typeName);
	}
	
	
	public void isBoundingBoxRelevant(boolean state) {
		vertex.fields().setBool8Field("isBoundingBoxRelevant", state);
	}
	
	
	/**
	 * 
	 * @param solidity
	 */
	public void setStrokeSolidity(WebSVG_StrokeSolidity solidity) {
		vertex.fields().setUInt8Field("strokeSolidity", solidity.code);
	}
	
	
	/**
	 * 
	 * @param color
	 */
	public void setStrokeColor(WebSVG_StrokeColor color) {
		vertex.fields().setUInt8Field("strokeColor", color.code);
	}
	
	
	/**
	 * Style: 
	 * @param thickness
	 */
	public void setStrokeThickness(float thickness) {
		vertex.fields().setFloat32Field("strokeThickness", thickness);
	}

}
